### PR TITLE
Write Notifications YAML Reference

### PR DESCRIPTION
**Pull request by mactsouk**

### CODE REVIEW

File: notifications-yaml-reference_5be45bb02c7d3a31944dbc98.md [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+ * [Properties](#properties)
+ * [The rules property in more detail](#the-rules-property-in-more-detail)
+ * [See Also](#see-also)
+
+## Overview
+
+You can find out how to create new notifications by visiting the
+[Slack Notification](https://docs.semaphoreci.com/article/91-slack-notifications)
+and the [sem command line tool Reference](https://docs.semaphoreci.com/article/53-sem-reference)
+pages of the Semaphore 2.0 documentation.
```
Comment (shiroyasha) [inline notifications-yaml-reference_5be45bb02c7d3a31944dbc98.md]: The title `Overview` doesn't match the content of the paragraph.

File: notifications-yaml-reference_5be45bb02c7d3a31944dbc98.md [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+
+## Rules of Notifications
+
+Notifications are governed by certain rules:
+
+* Notifications can be sent only when a pipeline is finished. Notifications for
```
Comment (shiroyasha) [inline notifications-yaml-reference_5be45bb02c7d3a31944dbc98.md]: Its a bit hard to follow the docs when the rules describe what is not possible instead of what is possible.

I propose the following:

- First describe what is possible in several bulletpoints
- Then, we can add a paragraph that exaplains what is not posssible. 

Also, I am not sure if we should concentrate too much on what is not possible, it highlights the negative parts of the softaware instead of the positive ones.

File: notifications-yaml-reference_5be45bb02c7d3a31944dbc98.md [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+begin the debugging process by verifying the Slack WebHook used.
+
+Tip: you can use just a single Incoming WebHook from Slack for all your
+notifications as this Incoming WebHook has access to all the channels of a
+Slack Workspace.
+
+##### message
+
+The `message` property allows you to define the message that will be displayed
+on the Slack channel or the Slack user by the rule of the notification.
```
Comment (shiroyasha) [inline notifications-yaml-reference_5be45bb02c7d3a31944dbc98.md]: This is not supported yet. Let's remove it.

File: notifications-yaml-reference_5be45bb02c7d3a31944dbc98.md [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+The second rule is more complex than the first one. The values of the `projects`
+property specify two exact project names named `S1` and `docs` as well as a
+regular expression that will be evaluated against all the project names of the
+current organization.
+
+Similarly, the `branches` property has two items. The first one is a regular
+expression that matches all branches that contain the `mt` sequence of
+characters and the second one is an exact match to the `master` branch.
+
+After than, you specify in the `pipelines` filter that you are only interested
```
Comment (shiroyasha) [inline notifications-yaml-reference_5be45bb02c7d3a31944dbc98.md]: Is "After than" a typo?

File: notifications-yaml-reference_5be45bb02c7d3a31944dbc98.md [hunk 5]
```diff
@@ -0,0 +1,207 @@
...
+## An example
+
+The following YAML code present an example of a notification as returned by the
+`sem get notification [name]` command:
+
+     sem get notifs docs
+    apiVersion: v1alpha
+    kind: Notification
+    metadata:
+      name: docs
```
Comment (shiroyasha) [inline notifications-yaml-reference_5be45bb02c7d3a31944dbc98.md]: If this notification is sending notifications for other projects appart from the docs project, I would suggest changing this name to something that is demonstrating the value of describing multiple independent rules in one resource.

For example, `documentation-release-cycle` would be a good notification name examples for a notification resource that describes various aspects of the documentation projects.

However, I would push it in the direction of using common scenarios for software release. Documentation is not the most common use case of notifications. Notifications for releasing and notifying multiples team for new API versions is much more common and instantly recognizable.

**Comment by shiroyasha** [comment]
Overall, I like the documentation page 👍 

I would put more emphasis on common real world examples. From my experience most of the readers will use examples as their first jumping point in reading the docs, and read the rest of the document only if the examples are too complex or hard to understand.

Optimizing for common use cases is a huge win for learning a software more easily.

File: notifications-yaml-reference_5be45bb02c7d3a31944dbc98.md (hunk 2 above)
Comment (mactsouk) [reply notifications-yaml-reference_5be45bb02c7d3a31944dbc98.md]: Good poing.

File: notifications-yaml-reference_5be45bb02c7d3a31944dbc98.md (hunk 3 above)
Comment (mactsouk) [reply notifications-yaml-reference_5be45bb02c7d3a31944dbc98.md]: Removed mentions to `message`.

File: notifications-yaml-reference_5be45bb02c7d3a31944dbc98.md [hunk 6]
```diff
@@ -0,0 +1,207 @@
...
+
+Similarly, the `branches` property has two items. The first one is a regular
+expression that matches all branches that contain the `mt` sequence of
+characters and the second one is an exact match to the `master` branch.
+
+After than, you specify in the `pipelines` filter that you are only interested
+in the main pipeline that begins with a `semaphore.yml` file only.
+
+Once these three filters are evaluated to `true`, the rule will send a
+notification to the `#dev-null` channel.
```
[the remark this answers is not in the document]
Comment (mactsouk) [reply notifications-yaml-reference_5be45bb02c7d3a31944dbc98.md]: Good point.

File: notifications-yaml-reference_5be45bb02c7d3a31944dbc98.md [hunk 7]
```diff
@@ -0,0 +1,207 @@
...
+        filter:
+          projects:
+          - docs
+        notify:
+          slack:
+            endpoint: https://hooks.slack.com/services/XXTXXSSA/ABCDDAS/XZYZWAFDFD
+            channels:
+            - '#dev-null'
+            - '@mtsoukalos'
+      - name: Send notifications for S1
```
[the remark this answers is not in the document]
Comment (mactsouk) [reply notifications-yaml-reference_5be45bb02c7d3a31944dbc98.md]: Done.

File: notifications-yaml-reference_5be45bb02c7d3a31944dbc98.md [hunk 8]
```diff
@@ -0,0 +1,207 @@
...
+        filter:
+          projects:
+          - S1
+          - /.*-api$/
+          - docs
```
[the remark this answers is not in the document]
Comment (mactsouk) [reply notifications-yaml-reference_5be45bb02c7d3a31944dbc98.md]: Done.

File: notifications-yaml-reference_5be45bb02c7d3a31944dbc98.md [hunk 9]
```diff
@@ -0,0 +1,207 @@
...
+          branches:
+          - /.*mt/
```
[the remark this answers is not in the document]
Comment (mactsouk) [reply notifications-yaml-reference_5be45bb02c7d3a31944dbc98.md]: Done.

File: notifications-yaml-reference_5be45bb02c7d3a31944dbc98.md (hunk 5 above)
Comment (mactsouk) [reply notifications-yaml-reference_5be45bb02c7d3a31944dbc98.md]: That is true but the first example should be simple. I renamed it to `API-docs`.
